lib/segment/src/types.rs

```diff
@@ -1426,8 +1426,8 @@ impl FromStr for DateTimePayloadType {
         // - YYYY-MM-DD HH:MM
         // - YYYY-MM-DD
         // See: <https://github.com/qdrant/qdrant/issues/3529>
-        let datetime = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%dT%H:%M:%S")
-            .or_else(|_| chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d %H:%M:%S"))
+        let datetime = chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%dT%H:%M:%S%.f")
+            .or_else(|_| chrono::NaiveDateTime::parse_from_str(s, "%Y-%m-%d %H:%M:%S%.f"))
```

Comment:
Does this change means, that `2012-12-12 12:20:45` is not a valid date anymore?

Reply:
It is still valid, the last section allows 0 or more decimals

Reply:
In this case, could you please add new cases to the tests, and do not replace the existing ones

Reply:
Added the case of space-separated dates 👍